### ISSUE TITLE
error to warning

### ISSUE THREAD

**Issue by jarkkojs** [rust-lang/cargo#12232]
### Problem

This transcript emphasizes the problem:

```bash
$ cargo uninstall zellij
error: corrupt metadata, `/home/jarkko/.cargo/bin/zellij` does not exist when it should

$ touch .cargo/bin/zellij

$ cargo uninstall zellij
    Removing /home/jarkko/.cargo/bin/zellij

$ cargo uninstall zellij
error: package ID specification `zellij` did not match any packages
```

Why is exiting empty file better than non-existing file? Since cargo does not care of the contents, why is it logical to fail ENOENT case?

### Possible Solution(s)

Change error to warning and proceed on removing the meta-data.

**Comment by epage** [comment]
If I'm understanding correctly, this is only happening if you do:
```console
$ cargo install zellij
$ rm ~/.cargo/bin/zellij
$ cargo uninstall zellij
```

Is that correct?

A benefit I see to this is it would allow someone to clean up their install-metadata

I can see us being extra cautious at removing any files if one of the expected files isn't there.  Deleting is inherently destructive and caution should be exercised.

The part that isn't to clear to me though is what the use case is where this would happen enough (or even once) for this to be noticeable.

**Comment by LeSnake04** [comment]
I think it makes sense to add something like `cargo uninstall --force/-f` for cases like this. I got this error trying to clean some unused packages, so deleting the metadata and leftover dirs would make sense in that case.

**Comment by xpe** [comment]
> A benefit I see to this is it would allow someone to clean up their install-metadata

Yes, I found this issue after I manually deleted a binary from `~/.cargo/bin/`, could not `cargo uninstall`, and started looking at `~/.cargo/.cargo.toml` and `~/.cargo/bin/.crates2.json`. I realized I didn't want to "play doctor" with Cargo's innards. So now I'm here. :)

I propose three new sub-subcommands:

```console
cargo install --metadata
cargo install --metadata-check
cargo install --metadata-repair
```

I think I also support the suggestion of @LeSnake04 above: `cargo uninstall --force/-f` at the risk of sapping a bit of momentum around metadata-decrufting.

**Comment by jarkkojs** [comment]
> If I'm understanding correctly, this is only happening if you do:
> 
> $ cargo install zellij
> $ rm ~/.cargo/bin/zellij
> $ cargo uninstall zellij
> Is that correct?
> 
> A benefit I see to this is it would allow someone to clean up their install-metadata
> 
> I can see us being extra cautious at removing any files if one of the expected files isn't there. Deleting is inherently destructive and caution should be exercised.
> 
> The part that isn't to clear to me though is what the use case is where this would happen enough (or even once) for this to be noticeable.

Sorry, I've even forgot writing this but yes that was the pattern :-)
